sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -162,7 +154,60 @@ object GpuCast extends Arm {
 
   val INVALID_FLOAT_CAST_MSG: String = "At least one value is either null or is an invalid number"
 
-  def sanitizeStringToFloat(input: ColumnVector, ansiEnabled: Boolean): ColumnVector = {
+  def sanitizeStringToDecimal(
+      input: ColumnView,
+      ansiEnabled: Boolean): ColumnVector = {
+
+    // This regex gets applied to filter out known edge cases that would result in incorrect
```

Comment:
What are the known edge cases? would be very nice to know what we have to include this as it is expensive. Looking at the code it appears that is_fixed_point cuts off early if it sees something that it does not expect, so it might be nice to have a follow on issue to actually fix that, either in CUDF or in Spark specific code.

Reply:
Where is it cutting off early? 

Are you saying if I pass c = ["", "1.2", "3", ""] and if the boolean vector is initialized to true
d = c.is_fixed_point() = [false, true, true,  true]

basically everything after the first value in `d` is bogus?

Reply:
Sorry I was wrong. Reading through the code it looked like the check ignored anything after it saw something it didn't expect, but that is not true.

It looks like "1.5ABC" will result in a false being returned. Which if that is true, then I don't think we need the regular expression check at all any more.  That is what triggered this? Why do we need the regexp. What "edge cases" does it cover that are not covered by the existing type check code?

Reply:
You are right we don't need the regex check anymore as the cudf is reporting everything we need. This is relevant in case of a float because it needs to convert the "infinity" => "inf"